src/test/java/com/google/songgame/YoutubeParserTest.java

```diff
@@ -0,0 +1,102 @@
+package com.google.songgame.data;
+
+import org.junit.Assert;
+import org.junit.rules.ExpectedException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.Rule;
+import org.junit.Ignore;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class YoutubeParserTest {
+
+  private YoutubeParser youtubeParser;
+
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Before
+  public void init() {
+    youtubeParser = new YoutubeParser();
+  }
+
+  // Test getPlaylistIdFromUrl
```

Comment:
Instead of these comments, one convention for test method names you could use is "methodUnderTest_state"

So your first method would be called getPlaylistIdFromUrl_playlistUrlWithParameters()

Later on, your first test for extractVideoIdFromJson could be called extractVideoIdFromJson_nonIdData()

See the last sentence in https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names